### ISSUE TITLE
pm2 logging takes free space on AWS EC2 instance

### ISSUE THREAD

**Issue by irenashemesh** [Unitech/pm2#2997]
Tried to modify pm2 logging according to http://pm2.keymetrics.io/docs/usage/log-management/:

by running the app with the following scripts in package.json:

"start": "node ./node_modules/.bin/pm2 start index.js",
"poststart": "node ./node_modules/.bin/pm2 logs --output \"/dev/null\" --error \"/dev/null\"",

This did not help much - after a day 1.5GB of logging.

There is a section in this document named "JSON way", however it is not clear where (which file) this JSON should be used.

We need to disable logs, not manage via logrotate.

```
PM2 version: `PM2 2.5.0`
Node version: `6.10.0`
Linux
```

**Comment by vmarchaud** [comment]
Is the problem applications logs or PM2 logs ? Because it should not output any log with your current configuration.
The "JSON way" is just using what we call a [process file](http://pm2.keymetrics.io/docs/usage/application-declaration/), it's just a json file where you put your application configuration

**Comment by Unitech** [comment]
Have you tried to disable the logs: http://pm2.keymetrics.io/docs/usage/application-declaration/#disabling-logs

**Comment by Unitech** [comment]
You can try the --disable-logs option:

```
$ npm install Unitech/pm2#development -g
$ pm2 update
$ pm2 start app.js --disable-logs
```

**Comment by irenashemesh** [comment]
Hello,

Thank you for your reply and explanation about the "JSON way".

The problem is PM2 logs, under /tmp/.pm2/logs directory. There are 2 files,
one for errors and one for logs. The content of this directory gets to more
than 2GB in a matter of 3-4 days, was 1.5 GB after one day. There were no
errors - normal operation. This happened after running the app with the
configuration as described.

Regards,
*Irena Shemesh, **Software Developer*
Terem Technologies
(p) 02 9430 6877
(e) irena.shemesh@terem.com.au
(w) terem.com.au

Level 5, 90 Pitt St, Sydney NSW 2000.

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.

**Comment by irenashemesh** [comment]
Hi Alexandre,

Ok, I will try that.

Thank you!

Regards,

*Irena Shemesh, **Software Developer*
Terem Technologies
(p) 02 9430 6877
(e) irena.shemesh@terem.com.au
(w) terem.com.au

Level 5, 90 Pitt St, Sydney NSW 2000.

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
$ pm2 start xxx.js --disable-logs
```

**Comment by irenashemesh** [comment]
Hi ,

Thank you for addressing the issue. I will try your suggestion.

I have opened #2997 <https://github.com/Unitech/pm2/issues/2997> and saw my
emails content, with full personal data posted there, event with this
remark "This email and any files transmitted with it are confidential and
intended solely for the use of the individual or entity to whom they are
addressed.". Please make sure *NOT* to post my personal information without
my permission on the public space. I have deleted these entries from there.

Regards,
*Irena Shemesh, **Software Developer*
Terem Technologies
(p) 02 9430 6877
(e) irena.shemesh@terem.com.au
(w) terem.com.au

Level 5, 90 Pitt St, Sydney NSW 2000.

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
